### ISSUE TITLE
Questions about training ldm

### ISSUE THREAD

**Issue by pokameng** [CompVis/latent-diffusion#195]
Hi
@asanakoy @pesser @crowsonkb @theRealSuperMario @to3i 
I am retraining the LDMs, but the samples are bad.
I want to konw, is the number of `batch size` will affect the sample progress and quality?
I  use `batch size`=36 and `gpus`=3, but you set `batch size`=48

My config is shown as:
model:
  base_learning_rate: 2.0e-06
  target: ldm.models.diffusion.ddpm.LatentDiffusion
  params:
    linear_start: 0.0015
    linear_end: 0.0195
    num_timesteps_cond: 1
    log_every_t: 200
    timesteps: 1000
    first_stage_key: image
    image_size: 64
    channels: 3
    monitor: val/loss_simple_ema
    unet_config:
      target: ldm.modules.diffusionmodules.openaimodel.UNetModel
      params:
        image_size: 64
        in_channels: 3
        out_channels: 3
        model_channels: 224
        attention_resolutions:
        # note: this isn\t actually the resolution but
        # the downsampling factor, i.e. this corresnponds to
        # attention on spatial resolution 8,16,32, as the
        # spatial reolution of the latents is 64 for f4
        - 8
        - 4
        - 2
        num_res_blocks: 2
        channel_mult:
        - 1
        - 2
        - 3
        - 4
        num_head_channels: 32
    first_stage_config:
      target: ldm.models.autoencoder.VQModelInterface
      params:
        ckpt_path: /latent-diffusion-main/models/ldm/lsun_beds256/model.ckpt
        embed_dim: 3
        n_embed: 8192
        ddconfig:
          double_z: false
          z_channels: 3
          resolution: 256
          in_channels: 3
          out_ch: 3
          ch: 128
          ch_mult:
          - 1
          - 2
          - 4
          num_res_blocks: 2
          attn_resolutions: []
          dropout: 0.0
        lossconfig:
          target: torch.nn.Identity
    cond_stage_config: __is_unconditional__
data:
  target: main.DataModuleFromConfig
  params:
    batch_size: 36
    num_workers: 15
    wrap: false
    train:
      target: ldm.data.lsun.LSUNBedroomsTrain
      params:
        size: 256
    validation:
      target: ldm.data.lsun.LSUNBedroomsValidation
      params:
        size: 256

lightning:
  callbacks:
    image_logger:
      target: main.ImageLogger
      params:
        batch_frequency: 5000
        max_images: 8
        increase_log_steps: False

  trainer:
    benchmark: True
My results are as follow:
![image](https://user-images.githubusercontent.com/47957015/204182836-012dface-cf0b-41f7-a778-e47bfe04f341.png)
![image](https://user-images.githubusercontent.com/47957015/204183066-4d39c38d-e7de-4f22-bf93-f1755d3eb470.png)

**Comment by louislee617** [comment]
> Hi @asanakoy @pesser @crowsonkb @theRealSuperMario @to3i I am retraining the LDMs, but the samples are bad. I want to konw, is the number of `batch size` will affect the sample progress and quality? I use `batch size`=36 and `gpus`=3, but you set `batch size`=48
> 
> My config is shown as: model: base_learning_rate: 2.0e-06 target: ldm.models.diffusion.ddpm.LatentDiffusion params: linear_start: 0.0015 linear_end: 0.0195 num_timesteps_cond: 1 log_every_t: 200 timesteps: 1000 first_stage_key: image image_size: 64 channels: 3 monitor: val/loss_simple_ema unet_config: target: ldm.modules.diffusionmodules.openaimodel.UNetModel params: image_size: 64 in_channels: 3 out_channels: 3 model_channels: 224 attention_resolutions: # note: this isn\t actually the resolution but # the downsampling factor, i.e. this corresnponds to # attention on spatial resolution 8,16,32, as the # spatial reolution of the latents is 64 for f4 - 8 - 4 - 2 num_res_blocks: 2 channel_mult: - 1 - 2 - 3 - 4 num_head_channels: 32 first_stage_config: target: ldm.models.autoencoder.VQModelInterface params: ckpt_path: /latent-diffusion-main/models/ldm/lsun_beds256/model.ckpt embed_dim: 3 n_embed: 8192 ddconfig: double_z: false z_channels: 3 resolution: 256 in_channels: 3 out_ch: 3 ch: 128 ch_mult: - 1 - 2 - 4 num_res_blocks: 2 attn_resolutions: [] dropout: 0.0 lossconfig: target: torch.nn.Identity cond_stage_config: **is_unconditional** data: target: main.DataModuleFromConfig params: batch_size: 36 num_workers: 15 wrap: false train: target: ldm.data.lsun.LSUNBedroomsTrain params: size: 256 validation: target: ldm.data.lsun.LSUNBedroomsValidation params: size: 256
> 
> lightning: callbacks: image_logger: target: main.ImageLogger params: batch_frequency: 5000 max_images: 8 increase_log_steps: False
> 
> trainer: benchmark: True My results are as follow: ![image](https://user-images.githubusercontent.com/47957015/204182836-012dface-cf0b-41f7-a778-e47bfe04f341.png) ![image](https://user-images.githubusercontent.com/47957015/204183066-4d39c38d-e7de-4f22-bf93-f1755d3eb470.png)

hello, have you solved your problem about bad reproduced results? And  i also account this problem when i trained ldm on CelebAHQ dataset.

**Comment by zhenjiaa** [comment]
hello,how to obtain these experimental analysis graphs above?

**Comment by zyinghua** [comment]
> hello,how to obtain these experimental analysis graphs above?

In your logs there should be a folder named testtube, and inside (whichever version) there should be a tf folder containing an event file, refer to that event file with command like "tensorboard --logdir=tf", make sure you have tensorboard installed beforehand. Hope this works for you as well :)
